src/Tablebot/Plugin/Discord.hs

```diff
@@ -59,6 +61,37 @@ sendMessage m t = do
     Left _ -> throw $ MessageSendException "Failed to send message."
     Right _ -> return ()
 
+-- | @sendReplyMessage@ sends the input message @t@ as a reply to the triggering message
+-- @m@. This returns an @Either RestCallErrorCode Message@ to denote failure or
+-- return the 'Message' that was just sent.
+sendReplyMessage ::
+  Message ->
+  Text ->
+  EnvDatabaseDiscord s ()
+sendReplyMessage m t = do
+  let ref = MessageReference (Just (messageId m)) Nothing Nothing False
+  res <- liftDiscord . restCall $ R.CreateMessageDetailed (messageChannel m) (R.MessageDetailedOpts t False Nothing Nothing Nothing (Just ref))
+  case res of
+    Left _ -> throw $ MessageSendException "Failed to send message."
+    Right _ -> return ()
+
+-- | @sendCustomReplyMessage@ sends the input message @t@ as a reply to a provided message id
+-- @m@. This returns an @Either RestCallErrorCode Message@ to denote failure or
+-- return the 'Message' that was just sent.
+-- @fail'@ indicates whether the message should still send if the provided message id is invalid
+sendCustomReplyMessage ::
+  Message ->
+  MessageId ->
+  Bool ->
+  Text ->
+  EnvDatabaseDiscord s ()
+sendCustomReplyMessage m mid fail' t = do
+  let ref = MessageReference (Just mid) Nothing Nothing fail'
+  res <- liftDiscord . restCall $ R.CreateMessageDetailed (messageChannel m) (R.MessageDetailedOpts t False Nothing Nothing Nothing (Just ref))
+  case res of
+    Left _ -> throw $ MessageSendException "Failed to send message."
+    Right _ -> return ()
+
```

Comment:
This all looks strong and works from brief testing (it's the reminder changes that are causing issues)